core/sail/base/src/main/java/org/eclipse/rdf4j/sail/base/SailSourceBranch.java

```diff
@@ -301,7 +309,7 @@ void merge(Changeset change) {
 			pending.remove(change);
 			if (isChanged(change)) {
 				Changeset merged;
-				changes.add(change);
+				changes.add(Changeset.simpleClone(change));
```

Comment:
Here is a place where we add a ChangeSet that will be closed later, so we make a clone to allow the original ChangeSet to be closed in the correct location.